src/features/device-logs/lib/supervisor.ts

```diff
@@ -9,7 +9,7 @@ const MAX_LOGS_PER_BATCH = 10;
 
 export class Supervisor {
 	public convertLogs(logs: SupervisorLog[]): DeviceLog[] {
-		if (logs.length > MAX_LOGS_PER_BATCH) {
+		if (Array.isArray(logs) && logs.length > MAX_LOGS_PER_BATCH) {
```

Comment:
I would prefer this check to be higher up when grabbing `req.body` rather than nested a couple of levels deep where we would ideally expect the typings to be respected